app/components/truncated_text_component.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class TruncatedTextComponent < ViewComponent::Base
+  def initialize(text = nil, render_check: true)
+    @text = text
+    @render_check = render_check
+  end
+
+  def text_content
+    return content if content.present?
+
+    simple_format(@text)
+  end
+
+  def render?
+    @render_check && (@text.present? || content.present?)
+  end
```

Comment:
(Lines 9-17) Unless you're seeing a future use of these, I'd consider removing them, since you're not using them now.